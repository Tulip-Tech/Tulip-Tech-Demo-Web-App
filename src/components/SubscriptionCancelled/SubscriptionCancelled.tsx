import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './SubscriptionCancelled.module.scss';

import Button from '#components/Button/Button';

type Props = {
  expiresDate: string;
  onClose: () => void;
};

const SubscriptionCancelled: React.FC<Props> = ({ expiresDate, onClose }: Props) => {
  const { t } = useTranslation('account');

  return (
    <div className={styles.SubscriptionCancelled}>
      <h2 className={styles.title}>{t('subscription_cancelled.title')}</h2>
      <p className={styles.paragraph}>{t('subscription_cancelled.message', { date: expiresDate })}</p>
      <Button label={t('subscription_cancelled.return_to_profile')} variant="outlined" onClick={onClose} fullWidth />
    </div>
  );
};

export default SubscriptionCancelled;
