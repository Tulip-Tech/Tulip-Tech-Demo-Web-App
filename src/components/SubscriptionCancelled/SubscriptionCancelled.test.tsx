import React from 'react';
import { render } from '@testing-library/react';

import SubscriptionCancelled from './SubscriptionCancelled';

describe('<SubscriptionCancelled>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<SubscriptionCancelled onClose={vi.fn()} expiresDate="12/12/2021" />);

    expect(container).toMatchSnapshot();
  });
});
