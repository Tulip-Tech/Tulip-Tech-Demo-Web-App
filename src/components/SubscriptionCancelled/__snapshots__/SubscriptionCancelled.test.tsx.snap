// Vitest Snapshot v1

exports[`<SubscriptionCancelled> > renders and matches snapshot 1`] = `
<div>
  <div>
    <h2
      class="_title_8d0ebd"
    >
      subscription_cancelled.title
    </h2>
    <p>
      subscription_cancelled.message
    </p>
    <button
      class="_button_f8f296 _default_f8f296 _outlined_f8f296 _fullWidth_f8f296"
    >
      <span>
        subscription_cancelled.return_to_profile
      </span>
    </button>
  </div>
</div>
`;
